### PR TITLE
Reshape and reinterpret tutorial.

### PR DESCRIPTION
**Pull request by mzient**
Signed-off-by: Michał Zientkiewicz <mzient@gmail.com>

#### Why we need this PR?
*Pick one, remove the rest*
- It adds new feature: Reshape and Reinterpret tutorial (because we want our functionality documented)

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     * N/A
 - Affected modules and functionalities:
     * N/A
 - Key points relevant for the review:
     * N/A
 - Validation and testing:
     * N/A
 - Documentation (including examples):
     * What else?

**JIRA TASK**: DALI-1916

### CODE REVIEW

**Comment by mzient** [comment]
!build

**Review by JanuszL — APPROVED** [review]

**Comment by mzient** [comment]
!build

File: docs/examples/general/reinterpret.ipynb [hunk 1]
```diff
@@ -0,0 +1,637 @@
+{
```
Comment (jantonguirao) [inline docs/examples/general/reinterpret.ipynb:1]: <p>as if it has -&gt; as it if had</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA/DALI/pull/2791/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (jantonguirao) [inline docs/examples/general/reinterpret.ipynb:1]: <p>some flattened -&gt; some flattened array (or tensor?)</p><p> </p><p>Whe using wildcards -&gt; When using wildcards</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA/DALI/pull/2791/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (jantonguirao) [inline docs/examples/general/reinterpret.ipynb:1]: <p>You can also insert new dimension -&gt; You can also insert a new dimension</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA/DALI/pull/2791/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (jantonguirao) [inline docs/examples/general/reinterpret.ipynb:1]: <p>coumns -&gt; columns</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA/DALI/pull/2791/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

**Review by jantonguirao — APPROVED** [review]

**Comment by mzient** [comment]
!build

**Pull request merged**